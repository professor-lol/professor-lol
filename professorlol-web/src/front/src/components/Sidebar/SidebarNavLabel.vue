<template>
  <SidebarNavItem :classes="classList.navItem">
    <a
      :class="classList.navLink"
      :href="url"
    >
      <i :class="classList.icon" /> {{ name }}
    </a>
  </SidebarNavItem>
</template>

<script>
import SidebarNavItem from './SidebarNavItem'
export default {
  name: 'SidebarNavLabel',
  components: {
    SidebarNavItem
  },
  props: {
    name: {
      type: String,
      default: ''
    },
    url: {
      type: String,
      default: '#'
    },
    icon: {
      type: String,
      default: 'fa fa-circle'
    },
    classes: {
      type: [String, Array, Object],
      default: ''
    },
    label: {
      type: Object,
      required: true,
      default: () => {}
    }
  },
  computed: {
    classList () {
      const classes = {
        navItem: ['hidden-cn', ...this.itemClasses],
        navLink: 'nav-label',
        icon: [
          this.icon ? this.icon : 'fa fa-circle',
          this.label.variant ? `text-${this.label.variant}` : '',
          this.label.class ? this.label.class : ''
        ].join(' ')
      }
      return classes
    },
    itemClasses () {
      const classes = this.classes
      return !classes ? [] : typeof classes === 'string' || classes instanceof String ? classes.split(' ') : Array.isArray(classes) ? classes : Object.keys(classes).filter(i=>classes[i])
    }
  },
}
</script>
