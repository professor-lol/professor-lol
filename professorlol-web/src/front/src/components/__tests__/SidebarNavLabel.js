import { mount } from '@vue/test-utils'
import SidebarNavLabel from "../Sidebar/SidebarNavLabel";

describe("SidebarNavLabel.vue", () => {
  // Inspect the raw component options
  test("should have default props", () => {
    const wrapper = mount(SidebarNavLabel, {
      propsData: {
        name: 'test',
        url: '',
        icon: 'fa fa-circle',
        classes: '',
        label: { variant: 'info' }
      }
    })
    expect(wrapper.props().name).toEqual('test')
    expect(wrapper.props().url).toBe('')
    expect(wrapper.props().icon).toBe('fa fa-circle')
  });
  it('has computed properties', () => {
    expect(typeof SidebarNavLabel.computed.classList).toBe('function')
    expect(typeof SidebarNavLabel.computed.itemClasses).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = mount(SidebarNavLabel, {
      propsData: {
        name: 'test',
        label: { variant: 'info' },
        icon: 'fa fa-circle'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.classes()).toContain('hidden-cn')
  })
});
