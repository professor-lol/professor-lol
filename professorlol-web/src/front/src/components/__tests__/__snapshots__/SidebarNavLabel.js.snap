// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarNavLabel.vue renders correctly 1`] = `
<li
  class="nav-item hidden-cn"
>
  <a
    class="nav-label"
    href="#"
  >
    <i
      class="fa fa-circle text-info "
    />
     test
  
  </a>
</li>
`;
